external/storm-kafka/src/jvm/org/apache/storm/kafka/KafkaSpout.java

```diff
@@ -188,6 +189,47 @@ public void declareOutputFields(OutputFieldsDeclarer declarer) {
         }
     }
 
+    @Override
+    public Map<String, Object> getComponentConfiguration () {
```

Comment:
looks to be quite a bit of logic in here. If it makes sense lets move most of it into KafkaUtils.java